### PR TITLE
Add a role metadata endpoint for clients

### PR DESCRIPTION
**Pull request by AlanCoding**
this looks like

Screenshot from 2024-03-25 16-48-01

We could give the full data for test_app:

```json
{
    "child_models": {
        "aap.immutabletask": {},
        "aap.instancegroup": {},
        "shared.organization": {
            "aap.cow": {},
            "aap.inventory": {},
            "aap.namespace": {
                "aap.collectionimport": {}
            },
            "aap.parentname": {},
            "aap.positionmodel": {},
            "shared.team": {},
            "aap.uuidmodel": {},
            "aap.weirdperm": {}
        }
    },
    "role_permissions": {
        "aap.collectionimport": [
            "aap.change_collectionimport",
            "aap.delete_collectionimport",
            "aap.view_collectionimport"
        ],
        "aap.cow": [
            "aap.say_cow",
            "aap.change_cow",
            "aap.delete_cow",
            "aap.view_cow"
        ],
        "aap.immutabletask": [
            "aap.cancel_immutabletask",
            "aap.view_immutabletask",
            "aap.delete_immutabletask"
        ],
        "aap.instancegroup": [
            "aap.change_instancegroup",
            "aap.delete_instancegroup",
            "aap.view_instancegroup"
        ],
        "aap.inventory": [
            "aap.update_inventory",
            "aap.change_inventory",
            "aap.delete_inventory",
            "aap.view_inventory"
        ],
        "aap.namespace": [
            "aap.change_namespace",
            "aap.delete_namespace",
            "aap.view_namespace",
            "aap.add_collectionimport",
            "aap.change_collectionimport",
            "aap.delete_collectionimport",
            "aap.view_collectionimport"
        ],
        "shared.organization": [
            "shared.change_organization",
            "shared.delete_organization",
            "shared.view_organization",
            "aap.update_inventory",
            "aap.add_inventory",
            "aap.change_inventory",
            "aap.delete_inventory",
            "aap.view_inventory",
            "aap.add_namespace",
            "aap.change_namespace",
            "aap.delete_namespace",
            "aap.view_namespace",
            "aap.add_collectionimport",
            "aap.change_collectionimport",
            "aap.delete_collectionimport",
            "aap.view_collectionimport",
            "shared.member_team",
            "shared.add_team",
            "shared.change_team",
            "shared.delete_team",
            "shared.view_team",
            "aap.say_cow",
            "aap.add_cow",
            "aap.change_cow",
            "aap.delete_cow",
            "aap.view_cow",
            "aap.add_uuidmodel",
            "aap.change_uuidmodel",
            "aap.delete_uuidmodel",
            "aap.view_uuidmodel",
            "aap.add_positionmodel",
            "aap.change_positionmodel",
            "aap.delete_positionmodel",
            "aap.view_positionmodel",
            "aap.I'm a lovely coconut",
            "aap.crack",
            "aap.add_weirdperm",
            "aap.change_weirdperm",
            "aap.delete_weirdperm",
            "aap.view_weirdperm",
            "aap.add_parentname",
            "aap.change_parentname",
            "aap.delete_parentname",
            "aap.view_parentname"
        ],
        "aap.parentname": [
            "aap.change_parentname",
            "aap.delete_parentname",
            "aap.view_parentname"
        ],
        "aap.positionmodel": [
            "aap.change_positionmodel",
            "aap.delete_positionmodel",
            "aap.view_positionmodel"
        ],
        "shared.team": [
            "shared.member_team",
            "shared.change_team",
            "shared.delete_team",
            "shared.view_team"
        ],
        "aap.uuidmodel": [
            "aap.change_uuidmodel",
            "aap.delete_uuidmodel",
            "aap.view_uuidmodel"
        ],
        "aap.weirdperm": [
            "aap.I'm a lovely coconut",
            "aap.crack",
            "aap.change_weirdperm",
            "aap.delete_weirdperm",
            "aap.view_weirdperm"
        ],
        "system": [
            "aap.add_collectionimport",
            "aap.view_collectionimport",
            "aap.change_collectionimport",
            "aap.delete_collectionimport",
            "aap.change_cow",
            "aap.view_cow",
            "aap.say_cow",
            "aap.add_cow",
            "aap.delete_cow",
            "aap.view_immutabletask",
            "aap.delete_immutabletask",
            "aap.add_immutabletask",
            "aap.cancel_immutabletask",
            "aap.add_instancegroup",
            "aap.change_instancegroup",
            "aap.delete_instancegroup",
            "aap.view_instancegroup",
            "aap.update_inventory",
            "aap.delete_inventory",
            "aap.change_inventory",
            "aap.view_inventory",
            "aap.add_inventory",
            "aap.add_namespace",
            "aap.view_namespace",
            "aap.change_namespace",
            "aap.delete_namespace",
            "shared.view_organization",
            "shared.add_organization",
            "shared.delete_organization",
            "shared.change_organization",
            "aap.add_parentname",
            "aap.change_parentname",
            "aap.view_parentname",
            "aap.delete_parentname",
            "aap.view_positionmodel",
            "aap.change_positionmodel",
            "aap.delete_positionmodel",
            "aap.add_positionmodel",
            "aap.view_uuidmodel",
            "aap.delete_uuidmodel",
            "aap.change_uuidmodel",
            "aap.add_uuidmodel",
            "aap.I'm a lovely coconut",
            "aap.change_weirdperm",
            "aap.crack",
            "aap.view_weirdperm",
            "aap.delete_weirdperm",
            "aap.add_weirdperm"
        ]
    }
}
```

In some cases, this might get a _little_ big, but already test_app is close to what AWX would do.

### CODE REVIEW

**Comment by AlanCoding** [comment]
Data dump here - https://gist.github.com/AlanCoding/cdb6e9d3ddce4ab7d25cb8001155740b

**Review by bhavenst — APPROVED** [review]
Looks and works great as far as I can tell!

File: ansible_base/rbac/api/views.py [hunk 1]
```diff
@@ -1,20 +1,79 @@
...
+
+
+class RoleMetadataView(AnsibleBaseDjangoAppApiView, GenericAPIView):
+    """General data about models and permissions tracked by django-ansible-base RBAC
+
+    Information from this endpoint should be static given a server version.
+    This reflects model definitions, registrations with the permission registry,
+    and enablement of RBAC features in settings.
+
+    child_models: Mappings that indicate what resource roles can confer permission to child resources
```
Comment (AlanCoding) [inline ansible_base/rbac/api/views.py]: remove

**Comment by AlanCoding** [comment]
I got cold feet about the name here, I renamed it from `role_permissions` to `allowed_permissions`.

That seems like it makes way more sense to me right now.

**Pull request merged**